### ISSUE TITLE
Warnings of test not wrapped in act since 9.3.2

### ISSUE THREAD

**Issue by joaopedrodcf** [testing-library/dom-testing-library#1275]
- `@testing-library/dom` version: 9.3.2 or greater
- Testing Framework and version: jest 29.7.0
  
- DOM Environment: jest-environment-jsdom 29.7.0
  

### Relevant code or config:

```js
    it('...' () => {
        const user = userEvent.setup();
        render(<ResetPasswordForm />);

        const newPasswordInput = screen.getByLabelText(...);

        // The test will start warning here as can be seen in the screenshot below
        await user.type(newPasswordInput, ...);
```

### What you did:

Upgraded from 9.3.1 to 9.3.2

### What happened:

The warnings started appearing all over the place

### Problem description:

Upgrading from `"@testing-library/dom": "9.3.1",` to `"@testing-library/dom": "9.3.2"  will start showing warnings in almost every test.
All the warnings are related with not wrapping in act.
<img width="975" alt="image" src="https://github.com/testing-library/dom-testing-library/assets/11476152/02cad5fa-4bf7-4c1d-9a6a-0285f7750207">

### Suggested solution:

I don't have suggested solution as I never checked the codebase of dom testing library, probably something changed from version 9.3.1 to 9.3.2

**Comment by MatanBobi** [comment]
Hi @joaopedrodcf.
The `act` error is from React Testing Library, is there any specific reason why you have DTL installed explicitly?
We have this issue that suggests it might be related to multiple DTL versions installed: https://github.com/testing-library/react-testing-library/issues/1216

**Comment by joaopedrodcf** [comment]
Wow, that was really quick @MatanBobi  :D 

Yeah, it was getting duplicated like this! After removing it, it works again 🥇 

I think I saw it in next documentation but tried to look again but can't find it anymore maybe they updated their docs 😆 

Gonna close the issue

**Comment by joaopedrodcf** [comment]
Actually found the reason @MatanBobi 

https://testing-library.com/docs/ecosystem-user-event/

It was here the I found it, basically this dependency is a peer dep of the  user-event:

<img width="975" alt="image" src="https://github.com/testing-library/dom-testing-library/assets/11476152/805e4988-b33b-4b35-8f65-2fff2b4d9241">

**Comment by joaopedrodcf** [comment]
Just to add more information for the issue:

<img width="1215" alt="image" src="https://github.com/testing-library/dom-testing-library/assets/11476152/59919a38-d488-4485-901f-3ee2cff84dd9">

<img width="1198" alt="image" src="https://github.com/testing-library/dom-testing-library/assets/11476152/3188916b-dc87-4ac3-9d86-09221e36e933">

It can be seen in the yarn.lock the 9.0.0 was resolved to 9.3.1 and then 9.3.3 was the one I installed manually

So the solution for me was to basically relly in "resolutions" ( overrides in npm ) to force the same version and not having a warning in doing yarn because of a missing peer dependency


// package.json
```js
{
    "devDependencies": {
        "@testing-library/dom": "9.3.3",
        "@testing-library/react": "14.0.0",
        "@testing-library/user-event": "14.5.1",
    },
    "resolutions": {
        "@testing-library/dom": "9.3.3"
    }
}
```

**Comment by joaopedrodcf** [comment]
Thank you so much 🙏
